cgrok/grokre.c

```diff
@@ -383,6 +383,22 @@ static void grok_capture_add_predicate(grok_t *grok, int capture_id,
   grok_capture_add(grok, gct, renamed_only);
 }
 
+/* Parse the group name, up to the max length, and try to parse it as a hex number.
+   Return 0 if the string is not a valid number, or 1 otherwise.*/
+static int grok_parse_capture_id(char* groupName, int maxLen, int *id) {
+	int i = 0;
+	for (i=0; i<maxLen; i++) {
+		if (groupName[i] == 0) {
+			break;
+		}
+		if (! ((groupName[i] >= '0' && groupName[i] <= '9') || (groupName[i] >= 'a' && groupName[i] <= 'f'))) {
```

Comment:
What about `groupName` characters that are capital letters? Should we allow those?